src/app/rpmostree-builtin-status.cxx

```diff
@@ -1081,6 +1093,13 @@ rpmostree_builtin_status (int             argc,
       else
         cached_update_node = json_node_new (JSON_NODE_NULL);
       json_builder_add_value (builder, cached_update_node);
+      json_builder_set_member_name (builder, "update-driver");
```

Comment:
Hmm, seems like the JSON is missing the update driver ID, no?

But actually, it's even easier than that: you can just pass the deserialized `GVariant` object directly like we do for `transaction` above.

Reply:
Right, I left that out because the update driver ID could potentially be empty if the transaction wasn't called with `RPMOSTREE_CLIENT_ID=testing-agent-id`. But on second thought, I think if we're calling the `--register-driver` option, we should make sure the driver supplies the update driver ID, and return an error if not, right?

Reply:
Yeah probably.  However...we should debate a little whether the "id" should be the central thing or whether the systemd unit should be.

Probably the axis here is around the MCO versus not - when talking to rpm-ostree from a Kubernetes pod you just get a not-human-readable auto-generated cgroup slice.  But when doing it from a host systemd service you get a nice e.g. `zincati.service`.  (It's a little less nice in the case of gnome-software but it's still not bad, e.g.:
```
[root@quicksilver ~]# journalctl -u rpm-ostreed|grep client.*software | head -1
Jan 11 13:53:41 quicksilver rpm-ostree[2691]: client(id:gnome-software dbus:1.110 unit:app-gnome-gnome\x2dsoftware\x2dservice-2334.scope uid:1000) added; new total=1
[root@quicksilver ~]#
```

Reply:
There's an interesting parallel to this debate which is: identifying software from the dpkg/RPM package name versus the systemd unit.

Reply:
Perhaps the middle ground is: if the systemd unit ends in `.service`, use it.  Otherwise, use the client ID.

Reply:
Are we planning to use the `register-driver` option on the MCO side too (i.e. the MCO will make the `register-driver` DBus call)? Right now the code prints both: `AutomaticUpdates: driven by $name ($systemd unit)`, and I imagine we could just store both these pieces of information the `--json` as well.

Reply:
If I understand correctly: So if we're enforcing that `register-driver` must have an `id`, then the `id` can be the "central thing" for now since it's probably guaranteed to be something useful. And then in a follow-up when we add the functionality to provide more information about the state of the driver, then we try to get that information from the systemd unit if the systemd unit ends in `.service`.

Reply:
Yeah I definitely think we should teach the MCO to use this.  

Agree that we should enforce `id` being provided with `--register-driver` and display it prominently.  The systemd unit can stay just in the journal logs for now.  (Or perhaps conditionally display it with the `-v` i.e. `opt_verbose` option)

Reply:
One thing worth noting is that `RPMOSTREE_CLIENT_ID` today only has an effect if the driver is using the CLI, and only if the CLI decides to register via `RegisterClient`. AFAICT, this is currently not the case for Zincati because it runs as a separate UID which [systemd doesn't consider `active`](https://github.com/coreos/rpm-ostree/blob/01e6c56415760def6c5b5f3d1d4939437f0f7009/src/app/rpmostree-dbus-helpers.cxx#L191-L205). E.g. from my FCOS server:

```
Jan 19 12:09:09 obelix rpm-ostree[2605951]: Initiated txn Deploy for client(dbus:1.5847 unit:zincati.service uid:980): /org/projectatomic/rpmostree1/fedora_coreos
```

(Notice how it doesn't have an `id:...`.)

So we'll have to tweak this. Also worth noting that the env var is just an override to the default `cli`, which isn't very useful (but would still pass through the `--register-driver` check of requiring an ID). And at the D-Bus level, providing a client ID means the client has to call `RegisterClient`, which so far hasn't been necessary (though it's clearly not a big deal to require it).

Hmm... I wonder if we should just leave the client ID as a thing we log to help debugging but otherwise not really promote it to the `status` and instead have `register-driver` take a human-readable string like "Zincati", or "Machine Config Operator". Something meaningful that users can just copy/paste into their search engine (which IMO is one of the primary goals of #1747). Then it's also obvious that it's required and it makes the API cleaner, because the option itself takes a value. (And maybe then we can have a special `<systemd_unit_description>` value or something which just tells rpm-ostree to [use the `Description`](https://www.freedesktop.org/wiki/Software/systemd/dbus/#unitobjects) of the associated systemd unit).

Anyway, don't mean to derail this, but worth discussing. :)

Reply:
Ohh I see. Yeah I haven't actually tested this with Zincati so I wasn't aware of that. In that case I agree we should make `--register-driver` take in an argument - a human readable string.

Reply:
Just noting:
> Also worth noting that the env var is just an override to the default `cli`

I tried playing around in the CLI with my builds of rpm-ostree and it seems that if I don't provide the env var, the ID will indeed be NULL.

Reply:
Ah wow, I guess https://github.com/coreos/zincati/commit/c214f3e9d0918fd21c0e246d6a2424dbec9df3b2 wasn't tested?  I had thought that was working.  

Mmmm...OK I guess we can just add a string to the driver.